packages/datastore/src/datastore.ts

```diff
@@ -256,30 +257,6 @@ class Datastore implements IIterable<Table<Schema>>, IMessageHandler, IDisposabl
     }
   }
 
-  processMessage(msg: Message): void {
-    switch (msg.type) {
-    // External messages:
-    case 'datastore-transaction':
-      const m = msg as Datastore.TransactionMessage;
-      this._applyTransaction(m.transaction);
-      break;
-
-    // Internal messages (posted from `this`):
-    case 'transaction-begun':
-      if (this._context.inTransaction) {
-        console.warn(
-          `Automatically ending transaction (did you forget to end it?): ${
-            this._context.transactionId
-          }`);
-        this.endTransaction();
-      }
-      break;
-    case 'queued-transaction':
-      this._processQueue();
-      break;
```

Comment:
Interesting, I sort of thought that the queueing would be mostly obseleted by the cemetery, but perhaps not. I can try to revert some of this so that these two cases are handled.

In what circumstances do you think we would get an incoming transaction while in an open state?

Reply:
If the user forgets to call endTransaction, tries to do stuff asynchronously, or builds a Rube Goldberg machine to shoot themselves in the foot.